### DISCUSSION TITLE
Real time communication (stream data from db on main-thread while updating on isolate using workmanager)

### DISCUSSION THREAD

**Discussion by zagori** [simolus3/drift#2311] [Q&A]
Hi all,

I have an issue in crating the following scenario and hope i can get some solution for it:

### Current implementation:
I have a stream on main-thread listening to changes in a TableA. Meanwhile, the [workmanager](https://pub.dev/packages/workmanager) started a process in isolate to update data in TableA.

### Problem:
The stream is not triggered after data is updated in isolate.

### Goal:
When data in TableA is updated in isolate, the main-thread stream should be able to receive the updated data.

Please advise if you have any solution to such case. Appreciate it in advance. Thanks.

**Comment by kuhnroyal** [comment]
Did you try https://drift.simonbinder.eu/docs/advanced-features/isolates/

**Comment by simolus3** [comment]
Query streams are a drift feature that is not provided by the underlying sqlite3 database. So if you open the same sqlite3 database file in two different database instances, their query streams are not synchronized.

You can open multiple instances of the same logical drift database with a mechanism described in the [isolates](https://drift.simonbinder.eu/docs/advanced-features/isolates/) documentation as @kuhnroyal pointed out. Essentially, the only thing you need is a two-way communication channel between the background task and the main isolate. How you implement that channel is up to you - drift has builtin support for a channel built with `SendPort` and `ReceivePort`, but any channel implementation can be used. You can even share a drift database across a network socket. If you have a way of making the background task talk to the UI isolate, you can also synchronize stream queries around that.

By skimming the readme of the `workmanager` package, it seems like there is no obvious way to do that unfortunately. I think there are other packages for background work that allow this, but I couldn't find anything for `package:workmanager`. If you at least have a way of knowing when a task was completed, you can update streams manually by calling something like this in your database class:

```dart
void updateAllStreams() {
  notifyUpdates({for (final table in allTables) TableUpdate.onTable(table)});
}
```
